### PR TITLE
[5.3] Add a link method to the filesystem, for symlinking

### PR DESCRIPTION
**Pull request by JosephSilber**
I manually tested this on Windows for both files and directories.

### CODE REVIEW

File: src/Illuminate/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -490,4 +508,14 @@ public function cleanDirectory($directory)
...
     }
+
+    /**
+     * Checks whether the system is running on Windows.
+     *
+     * @return bool
+     */
+    protected function isWindows()
+    {
+        return DIRECTORY_SEPARATOR == '\\';
```
Comment (jmarcher) [inline src/Illuminate/Filesystem/Filesystem.php]: Why not use PHP_OS ?
I don't agree that this works always

Comment (JosephSilber) [inline src/Illuminate/Filesystem/Filesystem.php]: PHP_OS is unreliable, and much more costly.

File: src/Illuminate/Filesystem/Filesystem.php [hunk 2]
```diff
@@ -490,4 +508,14 @@ public function cleanDirectory($directory)
     {
         return $this->deleteDirectory($directory, true);
     }
+
+    /**
+     * Checks whether the system is running on Windows.
+     *
+     * @return bool
+     */
+    protected function isWindows()
```
Comment (GrahamCampbell) [inline src/Illuminate/Filesystem/Filesystem.php]: Don't we already have an is windows function in support?

File: src/Illuminate/Filesystem/Filesystem.php (hunk 1 above)
Comment (GrahamCampbell) [inline src/Illuminate/Filesystem/Filesystem.php]: Please just use https://github.com/laravel/framework/blob/5.2/src/Illuminate/Support/helpers.php#L869.

Comment (taylorotwell) [inline src/Illuminate/Filesystem/Filesystem.php]: What Graham said ^

**Pull request merged**